agent/rpc_to.go

```diff
@@ -71,12 +72,48 @@ func (a *orbAgent) sendGroupMembershipReq() error {
 	if token := a.client.Publish(a.rpcToCoreTopic, 1, false, body); token.Wait() && token.Error() != nil {
 		return token.Error()
 	}
-
 	return nil
 }
 
+func (a *orbAgent) sendGroupMembershipReq() error {
+	defer a.retryGroupMembershipRequest()
+	return a.sendGroupMembershipRequest()
+}
+
+func (a *orbAgent) retryGroupMembershipRequest() {
+	if a.groupRequestTicker == nil {
+		a.groupRequestTicker = time.NewTicker(retryRequestFixedTime * retryRequestDuration)
+	}
+	go func() {
+		defer a.groupRequestTicker.Stop()
+		for calls := 1; calls <= retryMaxAttempts; calls++ {
+			select {
+			case <-a.groupRequestSucceeded:
+				a.groupRequestTicker.Stop()
+				return
+			case _ = <-a.groupRequestTicker.C:
+				a.logger.Info("agent has not received any group membership from fleet, re-requesting")
```

Comment:
let's display the timeout value in this message, i.e. how long we just waited unsuccessfully